// blind-signature/dist/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const sign_1 = require("./lib/sign");
const hash_1 = require("./lib/hash");
const blind_1 = require("./lib/blind");
var hash_2 = require("./lib/hash");
exports.longHash = hash_2.longHash;
var blind_2 = require("./lib/blind");
exports.getBlindingFactor = blind_2.getBlindingFactor;
function hashAndBlindMessage(key, message) {
    const messageHash = hash_1.longHash(key.n.bitLength(), message).toString('hex');
    return blind_1.blindMessageHash({
        messageHash,
        e: key.e,
        n: key.n
    });
}
exports.hashAndBlindMessage = hashAndBlindMessage;
function signBlindedMessageHash(key, blindedMessageHash) {
    return sign_1.sign(key, blindedMessageHash);
}
exports.signBlindedMessageHash = signBlindedMessageHash;
function unblindSignature(key, blindSignature, blindingFactor) {
    return blind_1.unblindBlindSignature({
        n: key.n,
        blindSignature,
        blindingFactor
    });
}
exports.unblindSignature = unblindSignature;
function verifySignature(key, message, signature) {
    return sign_1.verify(key, message, signature);
}
exports.verifySignature = verifySignature;
